### PR TITLE
p7zip: clarify license and remove unRAR code from src drv

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Description of changes

- p7zip: clarify license and remove unRAR code from src drv
- ~p7zip: add notice of the license updates~

In the same vein as #163999 and inspired by #163680 this tackles unRAR code

p7zip was already correctly building without unRAR code by default but since the
unRAR licensing also refers to the code being on your hard drive it's best to nuke
it from the src drv like in #163999 (and most of the changes were reusable from there) 

cc: @7c6f434c

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 7c6f434c** [comment]
Hm, is it a major or breaking change to need release notes? Presumably all compiled packages should behave the same as before…

**Comment by 06kellyjac** [comment]
This notice isn't in breaking changes, just in "Other Notable Changes". I can remove it if you want

The `_7zz` PR does have it in breaking changes since that one previously had unRAR code by default and now doesn't

**Comment by 7c6f434c** [comment]
I agree about _7zz but here the change is supposed to be hard to notice for the package user. I think it is better to remove.

**Review by wamserma — APPROVED** [review]
Build on aarch64-linux succeeded, diff LGTM:

**Comment by 06kellyjac** [comment]
OK ill remove the note tomorrow

**Comment by 06kellyjac** [comment]
Notice has been removed

File: pkgs/tools/archivers/p7zip/default.nix [hunk 1]
```diff
@@ -46,13 +52,20 @@ stdenv.mkDerivation rec {
 
   NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-error=c++11-narrowing";
 
-  meta = {
+  passthru.updateScript = ./update/update.sh;
```
Comment (7c6f434c) [inline pkgs/tools/archivers/p7zip/default.nix:55]: Hm, shouldn't that file be also among changes?

Comment (06kellyjac) [reply pkgs/tools/archivers/p7zip/default.nix:55]: :facepalm: 

added

**Pull request merged**